modules/tensor_mechanics/src/actions/GlobalStrainAction.C

```diff
@@ -125,24 +125,23 @@ GlobalStrainAction::act()
   //
   else if (_current_task == "add_aux_variable")
   {
+    MooseVariable & disp = _problem->getStandardVariable(0, _disp[0]);
```

Comment:
I'd get each displacement variable just to cover you bases in case somebody decides to use mixed displacement variable orders.

Reply:
...if that is possible...

Reply:
The order of ALL the displacement variables has to be of the same order as the mesh elements.  
https://github.com/idaholab/moose/blob/next/framework/src/problems/FEProblemBase.C#L5400

Reply:
Anyway, this does not work because 'GlobalStrainAction' is setting up the objects before the variables are set up. Its even called before the MasterAction. I believe the dependency between the two actions is not clear to MOOSE here. We probably should check the order from the mesh instead.

Reply:
Yes, check order from mesh. This is how I did it in the TM master action.